src/ezdxf/addons/drawing/pymupdf.py

```diff
@@ -358,7 +362,38 @@ def draw_filled_polygon(
     def draw_image(
         self, image: np.ndarray, transform: Matrix44, properties: BackendProperties
     ) -> None:
-        pass  # TODO: not implemented
+        height, width, depth = image.shape
+        assert depth == 4
+        corners = list(transform.transform_vertices([Vec2(0, 0), Vec2(width, 0), Vec2(width, height), Vec2(0, height)]))
+        xs = [p.x for p in corners]
+        ys = [p.y for p in corners]
+        r = fitz.Rect((min(xs), min(ys)), (max(xs), max(ys)))
+
+        # translation and non-uniform scale are handled by having the image stretch to fill the given rect.
+        angle = (corners[1] - corners[0]).angle_deg
+        need_rotate = not math.isclose(angle, 0.0)
+        # already mirroring once to go from pixels (+y down) to wcs (+y up)
+        # so a positive determinant means an additional reflection
+        need_flip = transform.determinant() > 0
+
+        if need_rotate or need_flip:
+            pil_image = PIL.Image.fromarray(image, mode='RGBA')
+            if need_flip:
+                pil_image = pil_image.transpose(PIL.Image.Transpose.FLIP_TOP_BOTTOM)
+            if need_rotate:
+                pil_image = pil_image.rotate(
+                    -angle,
+                    resample=PIL.Image.Resampling.BICUBIC,
+                    expand=True,
+                    fillcolor=(0, 0, 0, 0),
+                )
+            image = np.asarray(pil_image)
+            height, width, depth = image.shape
+
+        pixmap = fitz.Pixmap(fitz.Colorspace(fitz.CS_RGB), width, height, bytes(image.data), True)
+        # TODO: could improve by caching and re-using xrefs. If a document contains many
+        #  identical images redundant copies will be stored for each one
+        self.page.insert_image(r, keep_proportion=False, pixmap=pixmap)
```

Comment:
insert_image is quite limited in what it supports.
- It can only insert images that are axis-aligned (there is a rotation argument but must be a multiple of 90 degrees)
- the rect argument controls translation and scaling
- but with `keep_proportion=False` it can handle non-uniform scaling (but only axis-aligned)
- if the image has reflection or rotation I resample the image, for example (where white would be transparent):
![image282](https://github.com/mozman/ezdxf/assets/4923501/6a1052cc-55ff-4419-a7ca-c2b43ce10250)

the alternative approach would be to create a mini pdf document + page, insert the image then use `self.page.show_pdf_page()` which would avoid the need for resampling, but might be more tricky to get working